src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs

```diff
@@ -2983,5 +3007,51 @@ private static TimeZoneInfo CreateCustomLondonTimeZone()
                 }
             });
         }
+
+        //  This helper class is used to retrieve information about installed OS languages from Windows.
+        //  It should only be using in Windows platform-specific tests.
+        private static class WindowsUILanguageHelper
+        {
+            public static CultureInfo[] GetInstalledWin32CulturesWithUniqueLanguages() =>
+                GetInstalledWin32Cultures()
```

Comment:
nit: would it make sense to assert s_isWindows here? to ensure no one accidently call this method on other platforms?

Reply:
See my last update which changes it to return empty on other platforms, which allows me to also cache the call in a static, addressing the issue of calling it twice.